mlrun/datastore/base.py

```diff
@@ -132,6 +144,39 @@ def as_df(self, url, subpath, columns=None, df_module=None, format="", **kwargs)
                 kwargs["columns"] = columns
 
             def reader(*args, **kwargs):
+                if start_time or end_time:
+                    if sys.version_info >= (3, 7):
+                        from storey.utils import find_filters
+                    else:
+                        raise ValueError(
+                            f"feature not supported for python version {sys.version_info}"
+                        )
```

Comment:
A matter of style, but less indentation and code paths
```suggestion
                    if sys.version_info < (3, 7):
                        raise ValueError(
                            f"feature not supported for python version {sys.version_info}"
                        )
```